lms/djangoapps/bulk_email/models.py

```diff
@@ -168,16 +197,19 @@ def create(
 
         new_targets = []
         for target in targets:
+            # split target, to handle cohort:cohort_name
+            target_split = target.split(':', 1)
             # Ensure our desired target exists
-            if target not in EMAIL_TARGETS:
+            if target_split[0] not in EMAIL_TARGETS:
```

Comment:
Testing membership on a `list` takes O(N) time.  I see you're using the ordering of `EMAIL_TARGETS` to create `EMAIL_CHOICES`, but since that's all done at module-level anyway, you should make `EMAIL_TARGETS` a `set` which has O(1) access time for membership tests.